build.go

```diff
@@ -100,6 +100,10 @@ type BuildOptions struct {
 	// Buildpacks may both read and overwrite these values.
 	Env map[string]string
 
+	// Option only valid if Publish is true
```

Comment:
Is this a limitation of the lifecycle? It's likely not optimal for non-publish but I'm curious as to why it would be limited to publish == true.

Also, I don't see anywhere in our code where it is validated.

My preference is that this restriction isn't put in place unless there is a very technical reason why we should.

Reply:
Ya it's a limitation of the lifecycle implementation. It seems like we assume that any `-cache-image` arg passed to the lifecycle is a remote image [here](https://github.com/buildpacks/lifecycle/blob/ec698fb71333be8bf39440c8441864d21fe4587d/cache/image_cache.go#L33)

Reply:
Can we open an issue in lifecycle to add support for a local cache image, if there isn't one?

Reply:
Issue: https://github.com/buildpacks/lifecycle/issues/484